### PR TITLE
Feature/fee adjustment

### PR DESCRIPTION
**Pull request by stanieltron**
-adjusted buy and sell functions to deduct fee before tx can fail on min amount out / in 
-adjusted unit tests

QA: change in settle_treasury_buy_and_burn (simplified). Different results with sell and buy functions. Please check code and unit tests

### CODE REVIEW

**Comment by goncer** [comment]
@stanieltron  plz before merging and make this change definitive, 
Update the wiki here : 
https://github.com/mangata-finance/wiki/wiki/mangata-swap,-xyk-&-buy-and-burn
Feel free to paste my diagram there :)

and validate that the Table I created makes sense :)
https://github.com/mangata-finance/wiki/wiki/Mangata-fee-structure#tx-and-costs-relationship-table

You did a really great job simplifying the fees structure, so I bet it will crystal crlear!

File: pallets/xyk/src/lib.rs [hunk 1]
```diff
@@ -955,13 +940,17 @@ impl<T: Trait> XykFunctionsTrait<T::AccountId> for Module<T> {
         ensure!(first_asset_id != second_asset_id, Error::<T>::SameAsset,);
 
         // Liquidity token amount calculation
-        let initial_liquidity = first_asset_amount
-            .checked_add(second_asset_amount)
-            .ok_or_else(|| DispatchError::from(Error::<T>::MathOverflow))?;
+        let mut initial_liquidity = first_asset_amount / 2 + second_asset_amount / 2;
+        if initial_liquidity == 0 {
+            initial_liquidity = 1
```
Comment (gleb-urvanov) [inline pallets/xyk/src/lib.rs:945]: why not just +1?

File: pallets/xyk/src/lib.rs [hunk 2]
```diff
@@ -679,66 +721,17 @@ impl<T: Trait> Module<T> {
         Ok((first_asset_amount, second_asset_amount))
     }
 
+    //TODO if pool contains key !
```
Comment (gleb-urvanov) [inline pallets/xyk/src/lib.rs:724]: How critical is this TODO? Is it covered by e2e?

**Review by gleb-urvanov — APPROVED** [review]

File: pallets/xyk/src/lib.rs (hunk 1 above)
Comment (stanieltron) [reply pallets/xyk/src/lib.rs:963]: don't want to have +1, just a min of 1.
other reason is potential overflow max balance / 2 + max balance / 2 +1 = max balance +1

File: pallets/xyk/src/lib.rs (hunk 2 above)
Comment (stanieltron) [reply pallets/xyk/src/lib.rs:742]: unfortunate artifact from dev process, don't see any reason for this todo anymore

**Comment by stanieltron** [comment]
Wiki updated
@goncer Table is ok, just missing +1 after 0.05%, so its 0.05% of sold +1

**Pull request merged**